src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts

```diff
@@ -152,11 +157,44 @@ export class TableDesignerComponentInput implements DesignerComponentInput {
 			publishEvent.withAdditionalMeasurements({
 				'elapsedTimeMs': new Date().getTime() - startTime
 			}).withAdditionalProperties(metadataTelemetryInfo).send();
+			isPublishSuccessful = true;
 		} catch (error) {
 			this._errorMessageService.showDialog(Severity.Error, ErrorDialogTitle, localize('tableDesigner.publishChangeError', "An error occured while publishing changes: {0}", error?.message ?? error), error?.data);
 			this.updateState(this.valid, this.dirty);
 			this._adsTelemetryService.createErrorEvent(TelemetryView.TableDesigner, TelemetryAction.PublishChanges).withAdditionalProperties(telemetryInfo).send();
 		}
+
+		if (isPublishSuccessful) {
+			await this.refreshNodeInOE(isNewTable);
+		}
+	}
+
+	private async refreshNodeInOE(isNewTable: boolean) {
+		try {
+			const connectionId = this._objectExplorerContext.connectionProfile?.id;
+			const nodeInfo = this._objectExplorerContext.nodeInfo;
+			const node = await this._objectExplorerService.getTreeNode(connectionId, nodeInfo?.nodePath);
+			let refreshNodePath: string;
+
+			if (isNewTable) {
+				refreshNodePath = nodeInfo?.nodePath;
+			} else {
+				// This handle the case where a user publishes a table then edit the same table within the same designer then publish again. In that case node path in OE context is already correct.
+				if (node?.objectType === 'Tables') {
```

Comment:
This could go above as an || with isNewTable

Reply:
This is intentional as it's a tricky edge case and I think this way has better readability.